### PR TITLE
examinations

### PR DESCRIPTION
**Pull request by ngocvnn**

### CODE REVIEW

File: app/controllers/exams_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class ExamsController < ApplicationController
+  def index
+    @exam = Exam.new
+    @exams = current_user.exams.created_sort
+      .paginate page: params[:page], per_page: 5
+  end
+
+  def new
+
```
Comment (PhamVanTung) [inline app/controllers/exams_controller.rb]: remove line

File: app/controllers/exams_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  def new
+   
+  end
+
+  def create
+    @exam = Exam.new exam_params
+    if @exam.save
+      flash[:success] = "Examination created!"
```
Comment (PhamVanTung) [inline app/controllers/exams_controller.rb]: I18n

File: app/controllers/static_pages_controller.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class StaticPagesController < ApplicationController
   def home
+
```
Comment (PhamVanTung) [inline app/controllers/static_pages_controller.rb]: remove line

File: app/views/exams/_exam.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    <span> &nbsp; </span>
+  </li>
+  <br>
+  <li>
+    <span class="col-md-4">
+      <%= exam.category.name %>
+    </span>
+    <span class="col-md-4"> &nbsp;</span>
+    <span class="col-md-4">
+      <%= exam.status == "Start" ? (link_to "Start",'#') : (link_to "view",'#') %>
```
Comment (PhamVanTung) [inline app/views/exams/_exam.html.erb]: I18n

File: app/views/exams/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<% provide :title, "All examinations" %>
```
Comment (PhamVanTung) [inline app/views/exams/index.html.erb]: I18n

File: app/views/exams/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+<h1> All Examinations </h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @exam do |f| %>
+      <div class="border-c">
+        <%= f.collection_select :category_id, Category.all, :id, :name, 
+          {}, {class: "form-control", style: "width: auto"} %>    
+        <%= f.hidden_field :user_id, value: current_user.id %>
+        <%= f.hidden_field :status, value: "Start" %>
```
Comment (PhamVanTung) [inline app/views/exams/index.html.erb]: I18n

File: app/views/exams/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<% provide :title, "All examinations" %>
+<h1> All Examinations </h1>
```
Comment (PhamVanTung) [inline app/views/exams/index.html.erb]: I18n

File: app/views/exams/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for @exam do |f| %>
+      <div class="border-c">
+        <%= f.collection_select :category_id, Category.all, :id, :name, 
+          {}, {class: "form-control", style: "width: auto"} %>    
+        <%= f.hidden_field :user_id, value: current_user.id %>
+        <%= f.hidden_field :status, value: "Start" %>
+        <%= f.submit "Start new", class: "start-new btn btn-primary" %>
```
Comment (PhamVanTung) [inline app/views/exams/index.html.erb]: I18n

File: app/controllers/application_controller.rb [hunk 9]
```diff
@@ -3,6 +3,11 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :set_locale
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: remove

File: app/controllers/application_controller.rb [hunk 10]
```diff
@@ -3,6 +3,11 @@ class ApplicationController < ActionController::Base
...
+
+  def set_locale
```
Comment (dieunb) [inline app/controllers/application_controller.rb]: remove

File: db/seeds.rb [hunk 11]
```diff
@@ -15,13 +15,20 @@
   role: "admin"
 )
 
-19.times do |n|
+Category.create name: "MySQL", description: ""
```
Comment (dieunb) [inline db/seeds.rb]:
```
+Category.create name: "MySQL", description: ""
+Category.create name: "Ruby on Rails", description: ""
+Category.create name: "MySQL Exercise", description: ""
+Category.create name: "Git", description: ""
```
=>

```
["MySQL", "Ruby On Rails", "MySQL Exercise", "Git"].each do |name|
  Category.create name: name, description: ""
end
```

File: config/routes.rb [hunk 12]
```diff
@@ -1,9 +1,14 @@
 Rails.application.routes.draw do
-  root 'static_pages#home'
+  get 'home' => 'static_pages#home'
   get 'help' => 'static_pages#help'
   get 'about' => 'static_pages#about'
   get 'contact' => 'static_pages#contact'
 
   devise_for :users
   resources :users, only: :show
+  resources :exams
```
Comment (dieunb) [inline config/routes.rb]: filter routes

**Pull request merged**